File: sources/LiquidSense Release/me/aquavit/liquidsense/module/modules/ghost/AutoBow.java
package me.aquavit.liquidsense.module.modules.ghost;

import me.aquavit.liquidsense.LiquidSense;
import me.aquavit.liquidsense.event.EventTarget;
import me.aquavit.liquidsense.event.events.UpdateEvent;
import me.aquavit.liquidsense.module.Module;
import me.aquavit.liquidsense.module.ModuleCategory;
import me.aquavit.liquidsense.module.ModuleInfo;
import me.aquavit.liquidsense.value.BoolValue;
import net.minecraft.init.Items;
import net.minecraft.network.play.client.C07PacketPlayerDigging;
import net.minecraft.util.BlockPos;
import net.minecraft.util.EnumFacing;

@ModuleInfo(name = "AutoBow", description = "Automatically shoots an arrow whenever your bow is fully loaded.", category = ModuleCategory.GHOST)
public class AutoBow extends Module {

    private BoolValue waitForBowAimbot = new BoolValue("WaitForBowAimbot", true);
    
    @EventTarget
    public void onUpdate(final UpdateEvent event) {
        BowAimbot bowAimbot = (BowAimbot) LiquidSense.moduleManager.getModule(BowAimbot.class);
        if (mc.thePlayer.getHeldItem() != null && mc.thePlayer.getHeldItem().getItem() == Items.bow &&
                mc.thePlayer.isUsingItem() && mc.thePlayer.getItemInUseDuration() > 20 && (!waitForBowAimbot.get() || !bowAimbot.getState() || bowAimbot.hasTarget())) {
            mc.thePlayer.stopUsingItem();
            mc.getNetHandler().addToSendQueue(new C07PacketPlayerDigging(C07PacketPlayerDigging.Action.RELEASE_USE_ITEM, BlockPos.ORIGIN, EnumFacing.DOWN));
        }
    }
    
}

File: sources/Kevin Reborn/kevin/module/modules/combat/AutoBow.kt
/*
 * This program is free software: you can redistribute it and/or modify
 *   it under the terms of the GNU General Public License as published by
 *   the Free Software Foundation, either version 3 of the License, or
 *   (at your option) any later version.
 *
 *   This program is distributed in the hope that it will be useful,
 *   but WITHOUT ANY WARRANTY; without even the implied warranty of
 *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *   GNU General Public License for more details.
 *
 *   You should have received a copy of the GNU General Public License
 *   along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
package kevin.module.modules.combat

import kevin.event.EventTarget
import kevin.event.UpdateEvent
import kevin.main.KevinClient
import kevin.module.BooleanValue
import kevin.module.Module
import kevin.module.ModuleCategory
import net.minecraft.item.ItemBow
import net.minecraft.network.play.client.C07PacketPlayerDigging
import net.minecraft.util.BlockPos
import net.minecraft.util.EnumFacing

class AutoBow : Module("AutoBow", "Automatically shoots an arrow whenever your bow is fully loaded.", category = ModuleCategory.COMBAT) {
    private val waitForBowAimbot = BooleanValue("WaitForBowAimbot", true)

    @EventTarget
    fun onUpdate(event: UpdateEvent) {
        val bowAimbot = KevinClient.moduleManager.getModule(BowAimbot::class.java)

        val thePlayer = mc.thePlayer!!

        if (thePlayer.isUsingItem && thePlayer.heldItem?.item is ItemBow &&
            thePlayer.itemInUseDuration > 20 && (!waitForBowAimbot.get() || !bowAimbot.state || bowAimbot.hasTarget())) {
            thePlayer.stopUsingItem()
            mc.netHandler.addToSendQueue(C07PacketPlayerDigging(C07PacketPlayerDigging.Action.RELEASE_USE_ITEM, BlockPos.ORIGIN, EnumFacing.DOWN))
        }
    }
}